pyvoltha/adapters/kafka/kafka_proxy.py

```diff
@@ -317,20 +322,78 @@ def send_message(self, topic, msg, key=None):
             # port number.
             if self.stopping is False:
                 log.debug('stopping-kafka-proxy')
+                self.alive = False
                 try:
                     self.stopping = True
                     self.stop()
                     self.stopping = False
-                    self.faulty = False
                     log.debug('stopped-kafka-proxy')
                 except Exception as e:
                     log.exception('failed-stopping-kafka-proxy', e=e)
                     pass
             else:
                 log.info('already-stopping-kafka-proxy')
-
             return
 
+    # sending heartbeat message to check the readiness
+    def send_heartbeat_message(self, topic, msg):
+        assert topic is not None
+        assert msg is not None
+
+        try:
+            if self.kproducer_heartbeat is None:
+                if self.kafka_endpoint.startswith('@'):
+                    _k_endpoint = get_endpoint_from_consul(self.consul_endpoint,
+                                                           self.kafka_endpoint[
+                                                           1:])
+                else:
+                    _k_endpoint = self.kafka_endpoint
+
+                # Using 2 seconds timeout for heartbeat producer; default of 5 minutes is too long
+                self.kproducer_heartbeat = _kafkaProducer(
+                    {'bootstrap.servers': _k_endpoint,
+                     'default.topic.config' : {'message.timeout.ms': 2000},
+                    }
+                )
+
+            log.debug('sending-kafka-heartbeat-message', topic=topic)
+            msgs = [msg]
+
+            self.kproducer_heartbeat.produce(topic, msg, callback=self.handle_kafka_delivery_report)
+            log.debug('sent-kafka-heartbeat-message', topic=topic)
+
+        except Exception as e:
+            self.faulty = True
+            self.alive_state_handler.callback(self.alive)
+            log.error('failed-to-send-kafka-heartbeat-msg', e=e)
+
+    def check_heartbeat_delivery(self):
+        try:
+            if self.kproducer_heartbeat is not None:
+                log.debug('check-heartbeat-delivery')
```

Comment:
Remove this log and log 2 lines later. So much debug logs are not necessary; remember this poll is called every 2 seconds.